src/Config/TokenConfig.ts

```diff
@@ -0,0 +1,131 @@
+import MosaicConfig, { Address } from "./MosaicConfig";
+import {
+  InvalidTokenConfigException,
+  TokenConfigNotFoundException
+} from "../Exception";
+import * as fs from "fs-extra";
+import {Validator} from "jsonschema";
+const schema = require('./TokenConfig.schema.json');
+const path = require('path');
+
+/**
+ * Contract addresses of the origin chain specific to a token.
+ */
+export class OriginContracts {
+  public baseTokenAddress: Address;
+
+  public valueTokenAddress: Address;
+
+  public gatewayOrganizationAddress: Address;
+
+  public eip20GatewayAddress: Address;
+
+  public gatewayLibAddress?: Address;
+
+  public messageBusAddress?: Address;
+
+  public merklePatriciaLibAddress?: Address;
+
+  public stakePoolAddress?: Address;
+}
+
+/**
+ * Contract addresses deployed on the auxiliary chain specific to a token.
+ */
+export class AuxiliaryContracts {
+  public coGatewayOrganizationAddress: Address;
+
+  public utilityTokenAddress: Address;
+
+  public eip20CoGatewayAddress: Address;
+
+  public gatewayLibAddress?: Address;
+
+  public messageBusAddress?: Address;
+
+  public merklePatriciaLibAddress?: Address;
+
+  public redeemPoolAddress?: Address;
+}
+
+/**
+ * Holds the config of a deployed token.
+ */
+export default class TokenConfig {
+  public mosaicConfig: MosaicConfig;
+
+  public auxChainId: number;
+
+  public originContracts: OriginContracts;
+
+  public auxiliaryContracts: AuxiliaryContracts;
+
+  private constructor(config: any) {
+
+    this.mosaicConfig = MosaicConfig.fromFile(config.mosaicConfigFilePath);
+    this.auxChainId = config.auxChainId;
+    this.originContracts = config.originContracts;
+    this.auxiliaryContracts = config.auxiliaryContracts;
+  }
+
+  /**
+   * @param {string} filePath tokenConfig absolute path.
+   *
+   * @return {TokenConfig} TokenConfig object.
+   */
+  public static fromFile(filePath: string): TokenConfig {
+    if (fs.existsSync(filePath)) {
+      const configObject = TokenConfig.readConfigFromFile(filePath);
+      return new TokenConfig(configObject);
+    }
+    throw new TokenConfigNotFoundException(`Missing config file at path: ${filePath}`);
+  }
+
+  /**
+   * Read config from file, validate it and return as JSON object.
+   *
+   * @param {string} filePath tokenConfig absolute path.
+   *
+   * @return {object} Json parsed object.
+   */
+  private static readConfigFromFile(filePath: string): object {
```

Comment:
Do we need method to read token config based on chain identifier?

Reply:
I don't think so there is use case right now. Facilitator will read from tokenConfig file path.
We can add whenever the use case arises.

Reply:
Added TokenConfig.fromChain method.